pkg/cloudclient/aws/client/client.go

```diff
@@ -0,0 +1,64 @@
+package awsclient
+
+import (
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/aws/aws-sdk-go/service/sts/stsiface"
+)
+
+//go:generate mockgen -destination=mocks/client.go -package=mocks gitlab.cee.redhat.com/service/backplane-api/pkg/cloudclient/aws/client AwsClient
+
+// AwsClient is a representation of the upstream AWS client, as opposed to the
+// internal representation used by the rest of backplane. This is included in
+// the awsClient struct.
+type AwsClient interface {
+	// sts
+	AssumeRole(*sts.AssumeRoleInput) (*sts.AssumeRoleOutput, error)
+}
+
+// Client is a representation of the AWS Client
+type Client struct {
+	stsClient stsiface.STSAPI
+	AwsClient
+}
+
+// BuilderIface is a representation of the interface needed to create a builder
+type BuilderIface interface {
+	New(string, string, string, string) (AwsClient, error)
+}
+
+// Builder is a struct that returns a new AWS Client
+type Builder struct {
+	BuilderIface
+}
+
+// New is used when we already know the secrets and region, without any need to do any lookup.
+func (b *Builder) New(accessID, accessSecret, token, region string) (AwsClient, error) {
+	awsConfig := &aws.Config{
+		Region:                        aws.String(region),
+		CredentialsChainVerboseErrors: aws.Bool(true),
+	}
+	if token == "" {
+		os.Setenv("AWS_ACCESS_KEY_ID", accessID)
```

Comment:
Does this if statement make sense? I'm not sure

Reply:
yes I am wondering as well probably we can just use here the 
`credentials.NewStaticCredentials`
function as well

Reply:
We don't need to store the creds in ENV.